docs/timezone-draft.md

```diff
@@ -0,0 +1,169 @@
+> However, this must change, because implementations would need to call `super(id)` to set the _[[Identifier]]_ and _[[InitializedTemporalTimeZone]]_ internal slots.
+> Maybe we need to make `Temporal.TimeZone` not directly constructable?
+
+In order to perform attenuation, an implementation would monkeypatch the `Temporal.TimeZone.idToTimeZone()` function which performs the built-in mapping, and change the list of allowed time zones that `Temporal.TimeZone` iterates through.
+Or it could just replace the `Temporal.TimeZone` class altogether:
+
+```javascript
+// For example, to allow only offset time zones:
+
+TemporalTimeZone_idToTimeZone = Temporal.TimeZone.idToTimeZone;
+Temporal.TimeZone.idToTimeZone = function (id) {
+  if (/^[+-]\d{2}:?\d{2}$/.test(id))
+    return TemporalTimeZone_idToTimeZone(id);
+  return null;
+}
+Temporal.TimeZone[Symbol.iterator] = function* () { return null; }
+
+// or, to replace the built-in implementation altogether:
+
+Temporal.TimeZone = FrozenTimeZoneImplementation;
+```
+
+## Implementation of a custom time zone
+
+### TimeZone interface
+
+```javascript
+class Temporal.TimeZone {
+  /** Sets the [[Identifier]] internal slot to @id, and creates the
+   * [[InitializedTemporalTimeZone]] internal slot */
+  constructor(id : string) : Temporal.TimeZone;
+
+  // Methods that a subclassed custom time zone must implement
+
+  /** Given an instant (in epoch nanoseconds) returns this time zone's
+   * corresponding UTC offset, in nanoseconds (signed). */
+  getOffsetAtInstant(epochNs : bigint) : bigint;
+
+  /** Given the calendar/wall-clock time, returns an array of 0, 1, or
+   * 2 instants (in epoch nanoseconds) that are possible points on the
+   * timeline corresponding to it. */
+  possibleInstants(
+    isoYear : number,
+    isoMonth : number,
+    isoDay : number,
+    hour : number,
+    minute : number,
+    second : number,
+    millisecond : number,
+    microsecond : number,
+    nanosecond : number
+  ) : array<bigint>;
+
+  *getTransitions(startingPoint : Temporal.Absolute) : iterator<Temporal.Absolute>;
+
+  // API methods that a subclassed custom time zone doesn't need to touch
+
+  get name() : string;
+  getDateTimeFor(absolute : Temporal.Absolute) : Temporal.DateTime;
+  getAbsoluteFor(
+      dateTime : Temporal.DateTime,
+      options?: object
+  ) : Temporal.Absolute;
+  getOffsetFor(absolute : Temporal.Absolute) : string;
+  toString() : string;
+  toJSON() : string;
+
+  static from(item : any, options?: object) : Temporal.TimeZone;
+  static idToTimeZone(id: string) : Temporal.TimeZone;
+  static [Symbol.iterator]() : iterator<Temporal.TimeZone>;
+}
+```
+
+The `getOffsetAtInstant()` and `possibleInstants()` methods are conceptually "private", meaning that they are not intended as API to be normally used in user code.
```

Comment:
Why not make them public?  It seems like the feature they expose is potentially useful.

Reply:
You're quite right, come to think of it! `getOffsetForInstant()` would be a potential answer to #349, and `possibleInstants()` would be a potential answer to #318.